### PR TITLE
Select: add optgroup

### PR DESCRIPTION
**Pull request by seangates**
## Description
- adds an `<optgroup>` using a matched `optgroup` string

## References
Completes #689 

## Screenshots
59790957-ef068b80-9285-11e9-8b36-3dd98ad8f899

### CODE REVIEW

**Comment by DylanPiercey** [comment]
@ianmcburnie that is possible in Marko 3 and I'd recommend doing that now to avoid additional churn with the Marko 4 implementation.

**Comment by ianmcburnie** [comment]
@DylanPiercey I agree about the churn. However, Sean was having issues with nested tags. @seangates ?

**Comment by seangates** [comment]
@DylanPiercey Let me chat with you offline.

**Comment by seangates** [comment]
@ianmcburnie (et al),

After trying the additional markup, which more closely matches the markup of a native `<optgroup>`, it appears to not work with a `<ebay-select-option>` tag for both root level and inside of a `<ebay-select-optgroup>` tag.

It _does_ work if we were to have three sub tags `<ebay-select-option>`, `<ebay-select-optgroup>`, and `<ebay-select-optgroup-option>`. But I don't think we want to have the API be that inflexible.

Lastly, we certainly could use only two tags with Marko v4, though, as Dylan and I confirmed (another nail in v3's coffin).

In conclusion, the question becomes: do we want to implement this as is (the present code with an optgroup label, which works) and then change the API in Marko v4 (because we can do it better; e.g. the right way with just two tags), or do we punt it to Core v4 and do it correctly with just the two tags once Marko v4 components are in place?

**Comment by ianmcburnie** [comment]
@seangates I'm happy proceeding with what you have, in order to unblock selling team, and then we can change the API to be more what we would like in v4.

**Comment by seangates** [comment]
Sounds good. Proceeding.
